cache/network/download.go

```diff
@@ -48,12 +50,40 @@ func Download(ctx context.Context, params DownloadParams, logger log.Logger) (ma
 
 	logger.Debugf("Download archive")
 
-	downloadErr := downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
-	if downloadErr != nil {
-		return "", fmt.Errorf("failed to download archive: %w", downloadErr)
+	const maxRetries = 3
+	const retryDelay = 2 * time.Second
+	retriableErrors := []string{"Range request returned invalid Content-Length", "EOF"}
+
+	// Attempt to download, retrying on retriable errors
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		downloadErr := downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
+		if downloadErr == nil {
+			return restoreResponse.MatchedKey, nil
+		}
+
+		isRetriable := false
+		for _, retriableError := range retriableErrors {
```

Comment:
Could you add some tests? There is a lot of logic here for matching errors, handling context cancels, etc...